src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataComplexSerializer.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+#if NETFX // System.Data.Linq.Binary is only supported in the AspNet version.
+using System.Data.Linq;
+#endif
+using System.Diagnostics.Contracts;
+using System.Xml.Linq;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Represents an <see cref="ODataSerializer"/> for serializing <see cref="IEdmComplexType" />'s.
+    /// </summary>
+    public class ODataComplexSerializer : ODataEdmTypeSerializer
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ODataComplexSerializer"/>.
+        /// </summary>
+        public ODataComplexSerializer(ODataSerializerProvider serializerProvider)
+            : base(ODataPayloadKind.Resource, serializerProvider)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ODataEdmTypeSerializer"/> class.
+        /// </summary>
+        /// <param name="payloadKind">The kind of OData payload that this serializer generates.</param>
+        /// <param name="serializerProvider">The <see cref="ODataSerializerProvider"/> to use to write inner objects.</param>
+        protected ODataComplexSerializer(ODataPayloadKind payloadKind, ODataSerializerProvider serializerProvider)
+            : base(payloadKind, serializerProvider)
+        {
+            if (serializerProvider == null)
+            {
+                throw Error.ArgumentNull("serializerProvider");
+            }
```

Comment:
why do you check this here and not in the constructor on line 25?

Reply:
#Resolved